src/lib/logstore/log_store.cpp

```diff
@@ -283,13 +283,24 @@ void HomeLogStore::flush(logstore_seq_num_t upto_lsn) {
 }
 
 bool HomeLogStore::rollback(logstore_seq_num_t to_lsn) {
+    //Fast path
+    if (to_lsn == m_tail_lsn.load()) {
+	return true;
+    }
+
+    if (to_lsn > m_tail_lsn.load()) {
+        HS_LOG_ASSERT(false, "Attempted to rollback to {} which is larger than tail_lsn {}", to_lsn, m_tail_lsn.load());
+        return false;
+    }
+
```

Comment:
actually , what rollback does is to reset the tail_lsn to a valid lsn in the range [start_lsn , tail_lsn]. so if the rollback_to_lsn is not in the range, we can log an error and return false.
so my suggestion is add the following code is enough, and the check in line 297 can also be removed.
```
if(to_lsn < m_start_lsn || to_lsn > m_tail_lsn) {
        LOGERROR;
        return false;
}
```